packages/lesswrong/lib/collections/posts/schema.tsx

```diff
@@ -884,7 +891,16 @@ const schema: SchemaType<"Posts"> = {
       if (filteredTagRels?.length) {
         return filteredTagRels[0]
       }
-    }
+    },
+    sqlResolver: ({field, resolverArg, join}) => join({
+      table: "TagRels",
+      type: "left",
+      on: {
+        postId: field("_id"),
+        tagId: resolverArg("tagId"),
+      },
+      resolver: (tagRelField) => tagRelField("*"),
+    }),
```

Comment:
Initial thought, which may be resolved by reading further: Left joining will result in the thing on the left (the post in this case) being returned multiple times right (which will then have to be deduplicated)? I wonder if this might slow things down a lot esp for tables which have a large field (`contents`, even though it will soon be removed).

My knowledge of ORMs says that most of them fetch this kind of one to many relation in two queries (I know the Django ORM does this, and then Prisma does too but it's cursed wrt joins), which is closer to what the current dataloader setup does.

Anyway it's possible I've misunderstood this bc I've just started reviewing, so please correct me if this is wrong and it will actually not fetch duplicate posts

Reply:
Ok I was indeed misunderstanding this because I can see it's only returning one thing, mb